src/radio.cpp

```diff
@@ -30,99 +49,133 @@ namespace bzzz
     bool Radio::readPiData(void)
     {
         String allDataFromPi;
+        int data_count = 0;
 
         if (Serial.available() > 0)
         {
             allDataFromPi = Serial.readStringUntil('\n');
+            
+            if(allDataFromPi.substring(0, allDataFromPi.indexOf(",")) != "S") return false;
+            // cut the channelData string after the first occurence of a comma
+            allDataFromPi = allDataFromPi.substring(allDataFromPi.indexOf(",") + 1);
 
-            for (int i = 0; i < 16; i++)
+            for (int i = 0; i < NUM_RADIO_CHANNELS && allDataFromPi != ""; i++, data_count++)
```

Comment:
Why are you checking whether `allDataFromPi != ""`? Can this be the case? Would it be safer to check the length of `allDataFromPi` (e.g., whether it is too short)?

Reply:
That was my mistake. Fixed the logic. 
It was supposed to be
`for (int i = 0; i < NUM_RADIO_CHANNELS || allDataFromPi != ""; i++, data_count++)`

Here we check if we have consumed all of the string before we reach extract 8 data points. If yes, then we terminate the loop. Then this means that the `data_count != NUM_RADIO_CHANNELS`, so the function returns false. This is just an additional check to see if the data reception has been prematurely terminated. It could be the case that we receive a proper `S` at the beginning of the data, but the reception is interrupted for some reason before we read the entire string.